spmd/tensor/ops/dropout.py

```diff
@@ -0,0 +1,21 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+import torch
+from typing import Tuple
+from spmd.tensor.ops.utils import register_impl
+from spmd.tensor.api import Tensor
+
+
+@register_impl("aten.native_dropout.default")
+def _dist_dropout(self: Tensor, p: float, train: bool) -> Tuple[Tensor, Tensor]:
+    self_placement = self.placements[0]
+    # TODO: To figure out why partial tensor does not dispatch here when in CPU.
+    # and with kwargs.
+    if self_placement.is_partial():
```

Comment:
I think we should error out now for the replicated case, as it's non-deterministic for replicated output. We can enforce the seed generation to be the same for replicated tensor in later PRs

Reply:
Got it, sure.